File: excpp/otc-agent/solana/otc-program/scripts/check-desk.cpp
#include "check-desk.hpp"
#include <iostream>
#include <stdexcept>

namespace elizaos {

std::future<void> check() {
    // NOTE: Auto-converted from TypeScript - may need refinement

    const auto provider = AnchorProvider.env();
    setProvider(provider);
    const auto program = workspace.Otc<Otc>;

    const auto desk = new PublicKey("7EN1rubej95WmoyupRXQ78PKU2hTCspKn2mVKN1vxuPp");
    const auto data = program.account.desk.fetch(desk);

    std::cout << "Full desk data:" << /* JSON.stringify */ std::string(data, (_, v) => typeof v == "bigint" ? std::to_string(v) : v, 2) << std::endl;

}

} // namespace elizaos

File: otc-agent/solana/otc-program/scripts/check-desk.ts
import type { Program } from "@coral-xyz/anchor";
import pkg from "@coral-xyz/anchor";
import { PublicKey } from "@solana/web3.js";
import type { Otc } from "../target/types/otc";

// ESM/CJS compatibility: import as default then destructure
const { AnchorProvider, setProvider, workspace } = pkg as typeof import("@coral-xyz/anchor");

async function check() {
  const provider = AnchorProvider.env();
  setProvider(provider);
  const program = workspace.Otc as Program<Otc>;
  
  const desk = new PublicKey("7EN1rubej95WmoyupRXQ78PKU2hTCspKn2mVKN1vxuPp");
  const data = await program.account.desk.fetch(desk);
  
  console.log("Full desk data:", JSON.stringify(data, (_, v) => typeof v === 'bigint' ? v.toString() : v, 2));
}

check().catch(console.error);
